src/inmanta/data/__init__.py

```diff
@@ -4022,13 +4047,20 @@ async def save(self, connection: Optional[asyncpg.connection.Connection] = None)
 
     @classmethod
     async def purge_logs(cls) -> None:
-        environments = await Environment.get_list()
-        for env in environments:
-            time_to_retain_logs = await env.get(RESOURCE_ACTION_LOGS_RETENTION)
-            keep_logs_until = datetime.datetime.now().astimezone() - datetime.timedelta(days=time_to_retain_logs)
-            query = "DELETE FROM " + cls.table_name() + " WHERE environment=$1 AND started < $2"
-            value = cls._get_value(keep_logs_until)
-            await cls._execute_query(query, env.id, value)
+        default_retention_time = "7"
```

Comment:
@arnaudsjs I think this should do what we want but I don't know how to get the "default_retention_time" form [here](https://gist.github.com/FloLey/4e9b1883441a8bda3929ec9a820c6300#file-__init__-py-L2252).
Should I create a new global variable that would be used at that location and in the purge_logs function or is there another way?

Reply:
The `Environment` class in the database layer has an class variable `_settings` that contains all the settings and their default values. You can extract them directly from there:

```
Environment._settings[<name_environment_setting>].default
```
Does that make sense?